src/control/system/database.go

```diff
@@ -855,3 +857,38 @@ func (db *Database) UpdatePoolService(ps *PoolService) error {
 
 	return nil
 }
+
+// OnEvent handles events and updates system database accordingly.
+func (db *Database) OnEvent(_ context.Context, evt *events.RASEvent) {
+	switch evt.ID {
+	case events.RASPoolRepsUpdate:
```

Comment:
Wouldn't block on it, but if you have to make any more changes, it would probably make sense to break this logic out into its own method, both for testing and for code organization. I'm fine with waiting until a new event is added to this OnEvent() handler though.